### PR TITLE
*: Change ctx.Value from interface{} to Datum

### PR DESCRIPTION
**Pull request by shenli**
Use datum instead of interface for aggregate calculation.
@coocood @zimulala @BusyJay PTAL

### CODE REVIEW

File: evaluator/evaluator.go [hunk 1]
```diff
@@ -703,21 +703,21 @@ func (e *Evaluator) evalAggCount(v *ast.AggregateFuncExpr) {
...
-	switch x := ctx.Value.(type) {
-	case float64:
-		t := x / float64(ctx.Count)
-		ctx.Value = t
-		v.SetFloat64(t)
-	case mysql.Decimal:
+	switch ctx.Value.Kind() {
+	case types.KindFloat64:
+		t := ctx.Value.GetFloat64() / float64(ctx.Count)
+		v.SetValue(t)
```
Comment (zimulala) [inline evaluator/evaluator.go]: Here it can use `SetFloat64` directly？

Comment (shenli) [inline evaluator/evaluator.go]: v is ExprNode, not Datum.

**Comment by zimulala** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**